Common/MVKOSExtensions.mm

```diff
@@ -97,16 +97,18 @@ bool mvkGetEnvVarBool(std::string varName, bool* pWasFound) {
 #pragma mark System memory
 
 uint64_t mvkGetSystemMemorySize() {
+#if !MVK_TVOS
```

Comment:
@billhollings How do you feel about `!MVK_TVOS` usage here?

The `host_info` docs say available on macOS only: https://developer.apple.com/documentation/kernel/1502514-host_info?language=objc

However the headers say only prohibited on tvOS and watchOS:

```
__WATCHOS_PROHIBITED
__TVOS_PROHIBITED
kern_return_t host_info
{
```

Reply:
My thinking was that this would fail open, so when someone goes to port to watchos or catalyst, they'd see a compile failure if `host_info` is unavailable, instead of silently disabling this feature.

Reply:
Okay you've thoroughly convinced me. I like the idea of missing returns warnings as those were quite helpful when adding the tvOS branches.